### PR TITLE
tasks(partly)

### PR DESCRIPTION
**Pull request by vramaniuk**

### CODE REVIEW

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -356,9 +423,14 @@ function timespanToHumanString(startDate, endDate) {
...
  */
 function toNaryString(num, n) {
-    throw new Error('Not implemented');
+    var result = '';
+    while (num !== 0) {
+        result = (num % n) + result;
+        num = Math.floor(num / n);
+    }
+    return result;
 }
```
Comment (IlyaMokin) [inline task/06-conditions-n-loops-tasks.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toString

File: task/05-regex-tasks.js [hunk 2]
```diff
@@ -72,7 +73,7 @@ function getRegexForPitSpot() {
  * @return {RegExp}
  */
 function getRegexForIPv4() {
-   throw new Error('Not implemented');
+    return /^([0-9]|[0-9][0-9]|[01][0-9][0-9]|2[0-4][0-9]|25[0-5])(\.([0-9]|[0-9][0-9]|[01][0-9][0-9]|2[0-4][0-9]|25[0-5])){3}$/;
```
Comment (AlexMokin) [inline task/05-regex-tasks.js]: You can do it without repeating  the range part (0-255). Hint: use negative lookahead, nested groupts and notice that you can put $ at any place in your regex. This regex is good though. Had i seen this in production i wouldn't ask to fix it. But for a educational purpose try to do it without repeating.

Comment (vramaniuk) [reply task/05-regex-tasks.js]: my logic is to take all 3 digits numbers and exclude from match numbers from [256 to 999]
screenshot from 2017-09-19 12-37-44

/^(?!([2-9][6-9][0-9]|[3-9][0-9][0-9]))([0-9]|[0-9][0-9]|[0-9][0-9][0-9])(\.([0-9]|[0-9][0-9]|[0-9][0-9][0-9])){3}$/

File: task/06-conditions-n-loops-tasks.js (hunk 1 above)
Comment (vramaniuk) [reply task/06-conditions-n-loops-tasks.js]: Ok. It is a really simpler,but one usefull thing - I have bethought how numbers converted)

**Comment by vramaniuk** [comment]
So about __proto___ . Note accpted.  Solution might be 

function Rectangle(width, height) {
    this.width = width;
    this.height = height;
}

Rectangle.prototype.getArea = function() {
    return this.width * this.height;
};

**Comment by IlyaMokin** [comment]
done

**Pull request closed without merging**